### PR TITLE
ok, now we're using a recursive counter since indentation count was not ...

### PR DESCRIPTION
**Pull request by gidden**
...reliable due to conditions

required for cyclus/cycamore#297

### CODE REVIEW

File: cli/cycpp.py [hunk 1]
```diff
@@ -1255,17 +1256,20 @@ def read_vector(self, member, alias, t, uitype=None, ind='  ', idx=None):
...
+        s = ind + '{0}::InfileTree* bub = sub->SubTree("{1}", {2});\n'.format(
+            CYCNS, alias[0], tree_idx)
         s += ind + '{0}::InfileTree* sub = bub;\n'.format(CYCNS)
         s += ind + 'int n = sub->NMatches("{0}");\n'.format(alias[1])
         s += ind + '{0} {1};\n'.format(type_to_str(t), member)
         s += ind + '{0}.resize(n);\n'.format(member)
         s += ind + 'for (int {idx} = 0; {idx} < n; ++{idx})'.format(idx=idx) + ' {\n'
+        self._idx_lev += 1
         s += self.read_member('elem', alias[1], t[1], uitype[1], ind+'  ', idx=idx)
+        self._idx_lev -= 1
```
Comment (scopatz) [inline cli/cycpp.py]: Not blocking, but a context manager on the class would be a more elegant solution here:

```python
from contextlib import contextmanager

@contextmanager
def _nest_idx(self):
    self._idx_lev += 1
    yield
    self._idx_lev -= 1
```

Transforming this into 

```python
with self._nest_idx():
    s += self.read_member('elem', alias[1], t[1], uitype[1], ind+'  ', idx=idx)
```

**Comment by gidden** [comment]
I like it!

**Comment by gidden** [comment]
ping @scopatz, @rwcarlsen, this is currently blocking a number of cycamore PRs

**Comment by rwcarlsen** [comment]
Looks good - merging

**Pull request merged**